service/src/test/java/com/epam/esm/UserServiceTest.java

```diff
@@ -48,4 +68,24 @@ void testFindUserById() {
         verify(userRepository, times(1)).findById(id);
     }
 
+    @Test
+    void testFindAllWithPage() {
+        when(userRepository.findAllWithPage(anyInt(), anyInt())).thenReturn(List.of(new User(), new User()));
+
+        List<User> users = userService.findAllWithPage(1, 10);
+        assertEquals(2, users.size());
+        verify(userRepository, times(1)).findAllWithPage(anyInt(), anyInt());
+    }
+
+    @Test
+    void testGenerateToken() {
+        TokenRequest tokenRequest = new TokenRequest("testUser", "password");
+        when(authenticationManager.authenticate(any(UsernamePasswordAuthenticationToken.class))).thenReturn(null);
+        when(jwtTokenUtil.generateToken(anyString())).thenReturn(new TokenResponse("accessToken"));
+        TokenResponse tokenResponse = userService.generateToken(tokenRequest);
+
+        assertEquals("accessToken", tokenResponse.accessToken());
+        verify(authenticationManager, times(1)).authenticate(any(UsernamePasswordAuthenticationToken.class));
+        verify(jwtTokenUtil, times(1)).generateToken(anyString());
+    }
```

Comment:
Could you please add tests for exceptional cases also? For example, username maybe already exist or the password can be wrong.